### PR TITLE
doc: add resources chart

### PR DESCRIPTION
**Pull request by miroslavstastny**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Adds flamegrill resources charts to all perf charts in docsite:
image

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/fluentui/pull/12557)

### CODE REVIEW

File: packages/fluentui/docs/src/components/ComponentDoc/PerfChart/Chart.tsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+export type ChartDataSeries = {
+  name: string;
+  data: string;
+  color: string;
+  props?: object;
+};
+
+export type ChartProps = {
+  perfData: PerfData;
+  group: 'performance';
```
Comment (miroslavstastny) [inline packages/fluentui/docs/src/components/ComponentDoc/PerfChart/Chart.tsx:24]: I plan to add bundleSize in a follow-up PR.

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentPerfExample/ComponentPerfChart.tsx [hunk 2]
```diff
@@ -1,7 +1,12 @@
 import * as React from 'react';
 import { PerfChart, usePerfData, ComponentChart } from '../PerfChart';
+import { Checkbox } from 'src/index';
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentPerfExample/ComponentPerfChart.tsx:3]:
```suggestion
import { Checkbox } from '@fluentui/react-northstar';
```

File: packages/fluentui/docs/src/components/ComponentDoc/ComponentPerfExample/ComponentPerfExample.tsx [hunk 3]
```diff
@@ -14,16 +15,45 @@ export interface ComponentPerfExampleProps {
...
   const { title, description, examplePath } = props;
-  // FIXME: find a better way
+
+  const [currentChart, setCurrentChart] = React.useState<'perf' | 'resources'>('perf');
+  const onShowPerfChart = React.useCallback(() => {
+    setCurrentChart('perf');
+  }, []);
+  const onShowResourcesChart = React.useCallback(() => {
+    setCurrentChart('resources');
+  }, []);
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/ComponentPerfExample/ComponentPerfExample.tsx:25]:
```suggestion
  const [currentChart, setCurrentChart] = React.useState<'perf' | 'resources'>('perf');

  const onShowPerfChart = () => setCurrentChart('perf');
  const onShowResourcesChart = () => setCurrentChart('resources');
```

I don't see there any need to have `useCallback` here...

File: packages/fluentui/docs/src/components/ComponentDoc/PerfChart/ResourcesChart.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  return '00000'.substring(0, 6 - c.length) + c;
+}
+
+const ResourcesChart: React.FC<ResourcesChartProps> = ({ perfData, filter }) => {
+  const dataSeries: ChartDataSeries[] =
+    filter?.map(d => ({
+      name: d,
+      data: `flamegrill.profile.metrics.${d}`,
+      color: intToRGB(hashCode(d)),
+    })) || [];
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/PerfChart/ResourcesChart.tsx:44]:
```suggestion
const ResourcesChart: React.FC<ResourcesChartProps> = ({ perfData, filter = [] }) => {
  const dataSeries: ChartDataSeries[] =
    filter.map(d => ({
      name: d,
      data: `flamegrill.profile.metrics.${d}`,
      color: intToRGB(hashCode(d)),
    }));
```

I think this also should work

File: packages/fluentui/docs/src/components/ComponentDoc/PerfChart/ResourcesChartTooltip.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import { Crosshair } from 'react-vis';
+import { PerfSample } from './PerfDataContext';
+
+const values = [
```
Comment (layershifter) [inline packages/fluentui/docs/src/components/ComponentDoc/PerfChart/ResourcesChartTooltip.tsx:6]: Under `ResourcesChartProps` we have `filter`, should extract that type and link it here:

```suggestion
const values: FlamegrillMetrics = [
```

??

**Pull request merged**